src/b-splines/cubic.jl

```diff
@@ -0,0 +1,235 @@
+a condition where:
+
+    -cm + cp = 0
+"""
+function prefiltering_system{T,TC}(::Type{T}, ::Type{TC}, n::Int,
+                                   ::Type{Cubic{Flat}}, ::Type{OnGrid})
+    dl, d, du = inner_system_diags(T, n, Cubic{Flat})
+    d[1] = d[end] = -one(T)
+    du[1] = dl[end] = zero(T)
+
+    # Now Woodbury correction to set `[1, 3], [n, n-2] ==> 1`
+    specs = _build_woodbury_specs(T, n, (1, 3, one(T)), (n, n-2, one(T)))
+
+    Woodbury(lufact!(Tridiagonal(dl, d, du), Val{false}), specs...), zeros(TC, n)
+end
+
+# TODO: check to see if above method is the same as Quadratic
+
+"""
+`Flat` `OnCell` amounts to setting `y_0'(x) = 0` at `x = -1/2`. Applying this
+gives a condition where:
+
+    -9 cm + 11 c -3 cp + 1 cpp = 0
+"""
+function prefiltering_system{T,TC}(::Type{T}, ::Type{TC}, n::Int,
+                                   ::Type{Cubic{Flat}}, ::Type{OnCell})
+    dl, d, du = inner_system_diags(T,n,Cubic{Flat})
+    d[1] = d[end] = -9
+    du[1] = dl[end] = 11
+
+    # now need Woodbury correction to set :
+    #    - [1, 3] and [n, n-2] ==> -3
+    #    - [1, 4] and [n, n-3] ==> 1
+    specs = _build_woodbury_specs(T, n,
+                                  (1, 3, T(-3)),
+                                  (n, n-2, T(-3)),
+                                  (1, 4, one(T)),
+                                  (n, n-3, one(T))
+                                  )
+
+    Woodbury(lufact!(Tridiagonal(dl, d, du), Val{false}), specs...), zeros(TC, n)
+end
+
+"""
+`Line` `OnCell` amounts to setting `y_0''(x) = 0` at `x = -1/2`. Applying this
+gives a condition where:
+
+    3 cm -7 c + 5 cp -1 cpp = 0
+"""
+function prefiltering_system{T,TC}(::Type{T}, ::Type{TC}, n::Int,
+                                   ::Type{Cubic{Line}}, ::Type{OnCell})
+    dl,d,du = inner_system_diags(T,n,Cubic{Flat})
+    d[1] = d[end] = 3
+    du[1] = dl[end] = -7
+
+    # now need Woodbury correction to set :
+    #    - [1, 3] and [n, n-2] ==> -3
+    #    - [1, 4] and [n, n-3] ==> 1
+    specs = _build_woodbury_specs(T, n,
+                                  (1, 3, T(5)),
+                                  (n, n-2, T(5)),
+                                  (1, 4, -one(T)),
+                                  (n, n-3, -one(T))
+                                  )
+
+    Woodbury(lufact!(Tridiagonal(dl, d, du), Val{false}), specs...), zeros(TC, n)
+end
+
+function prefiltering_system{T,TC,GT<:GridType}(::Type{T}, ::Type{TC}, n::Int,
+                                                ::Type{Cubic{Periodic}}, ::Type{GT})
+    dl, d, du = inner_system_diags(T,n,Cubic{Periodic})
+
+    specs = _build_woodbury_specs(T, n,
+                                  (1, n, SimpleRatio(1, 6)),
+                                  (n, 1, SimpleRatio(1, 6))
+                                  )
+
+    Woodbury(lufact!(Tridiagonal(dl, d, du), Val{false}), specs...), zeros(TC, n)
+end
+
+
+#=
+NOTE: the following are the same as Quadratic and are implemented there:
```

Comment:
I would prefer if these were actually explicitly implemented here, but just forwarding to the implementations in quadratic.jl, e.g.

```
prefiltering_system{T,TC}(::Type{T}, ::Type{TC}, n::Int, ::Type{Cubic{Line}}, ::Type{OnGrid}) =
    prefiltering_system(T, TC, n, Quadratic{Line}, OnGrid)
```